File: packages/api/src/@core/magic-link/dto/create-magic-link.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class CreateMagicLinkDto {
  @ApiProperty({
    example: 'id_1',
    description: 'The linked user origin ID',
  })
  linked_user_origin_id: string;

  @ApiProperty({
    example: 'email@example.com',
    description: 'The email address',
  })
  email: string;

  @ApiProperty({
    example: '',
    description: 'The alias for the magic link',
  })
  alias: string;

  @ApiProperty({
    example: '801f9ede-c698-4e66-a7fc-48d19eebaa4f',
    description: 'The project ID',
  })
  id_project: string;
}

File: packages/sdk/output/python/src/testsdk/models/CreateMagicLinkDto.py
from .base import BaseModel


class CreateMagicLinkDto(BaseModel):
    def __init__(
        self,
        id_project: str,
        alias: str,
        email: str,
        linked_user_origin_id: str,
        **kwargs,
    ):
        """
        Initialize CreateMagicLinkDto
        Parameters:
        ----------
            id_project: str
            alias: str
            email: str
            linked_user_origin_id: str
        """
        self.id_project = id_project
        self.alias = alias
        self.email = email
        self.linked_user_origin_id = linked_user_origin_id
